ApollosProject/apollos-prototype
# Adds HorizontalContentCardConnected and Component prop to ContentCardConnected

## DESCRIPTION

This adds a `Component` prop to `ContentCardConnected` and changes `ContentCardConnected`s default rendered child to `DefaultCard`. Additionally, it adds a `HorizontalContentCardConnected` component. Lastly, we add some missing exports for some of the newer ui-kit cards 🙃 

### What does this PR do, or why is it needed?
This makes `ContentCardConnected` way more flexible by letting us pass in other cards to the same query. All of our news cards share the same same basic API so this is really easy 💥

Our new card designs also need to be rendered at the right time. This adds a mapper HOC that selects the right card for the right place.

### What design trade-offs/decisions were made?
TONS.

1. This PR breaks things. In particular horizontal card rendering is pretty broken. 
2. We also no longer render 1 card in two ways (vertical vs horizontal scroll). Instead we have separate components for that and there is fair amount of legacy `tile` props still floating around that probably could be refactored.
3. Merging this means committing otherwise broken snapshots to 😢
4. The new cards require images and in a lot of cases titles so there are a bunch of prop warnings.
5. no tests and no stories 😭

### How do I test this PR?
Check that the app now renders the new card styles in the feeds. This PR could use stories for `ContentCardConnected` and `HorizontalContentCardConnected` but they didn't exist before this so...

### What automated tests did you write?
This PR could use tests for `ContentCardConnected` and `HorizontalContentCardConnected` but they didn't exist before this so...

## TODO:

- [x] Partial refactor work associated with #852 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
